eclair-core/src/main/scala/fr/acinq/eclair/PluginParams.scala

```diff
@@ -55,4 +58,15 @@ trait CustomCommitmentsPlugin extends PluginParams {
    * returned by this method.
    */
   def getHtlcsRelayedOut(htlcsIn: Seq[IncomingHtlc], nodeParams: NodeParams, log: LoggingAdapter): Map[Origin, Set[(ByteVector32, Long)]]
+}
+
+// @formatter:off
+case class InterceptOpenChannelReceived(replyTo: ActorRef[InterceptOpenChannelResponse], open: Either[OpenChannel, OpenDualFundedChannel], temporaryChannelId: ByteVector32, localParams: LocalParams, fundingAmount: Satoshi)
+sealed trait InterceptOpenChannelResponse
+case class AcceptOpenChannel(temporaryChannelId: ByteVector32, localParams: LocalParams) extends InterceptOpenChannelResponse
+case class RejectOpenChannel(temporaryChannelId: ByteVector32, error: Error) extends InterceptOpenChannelResponse
+// @formatter:on
+
+trait InterceptOpenChannelPlugin extends PluginParams {
+  def getOpenChannelInterceptor: ActorRef[InterceptOpenChannelReceived]
```

Comment:
nit: getter/setter are so OOP, we usually omit them:
```suggestion
  def openChannelInterceptor: ActorRef[InterceptOpenChannelReceived]
```

Reply:
Fixed in [3c3a261](https://github.com/ACINQ/eclair/pull/2552/commits/3c3a2619e27a4ded6428ba3f587f1b9be223bff4)